### PR TITLE
Bug Fix: Same Type of Users Matched

### PR DESCRIPTION
**Pull request by mishasavlv**
# Pull Request

## Description

This branch fixes a bug where two adults were able to be matched due to a normalizeUserType function that defaulted to "student" for any input that did not match "adult". Therefore, if the NPO used the Registration Form Editor to rename the "Adult" option to any other label, the normalizeUserType function would default to output "student", causing the new user (adult) to erroneously register as a student. 

My solution was to add an optional field to the Question object called optionsLocked. If a multiple-choice question in the registration form editor has optionsLocked set to true, the options will not be editable.

With TA's approval, I also changed the registration form editor so that questions with locked set to true (not to be confused with my new field optionsLocked) were not able to have their required status toggled by the NPO admin. (Considering that if a question is locked by us, then an answer is probably needed and essential).

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from Notion)

Triage: Same type/admins matched (https://app.notion.com/p/h4i/Triage-Same-type-admins-matched-380197abf07b80be80edc06d46a61cbc?source=copy_link)

### Code Review Preparation
- [x] I have performed a self-review and reviewed the changed files
- [x] My code follows the project's style guidelines and passes linting
- [x] I have added comments in hard-to-understand areas and updated documentation

### Testing
- [x] I have added relevant tests and they pass locally
- [x] I have tested the changes manually, including edge cases

### Code Quality & Security
- [x] I have formatted my code and removed debugging/unused code
- [x] No TypeScript errors or linting issues
- [x] No sensitive information committed (API keys, passwords, etc.)
- [x] User inputs are validated and sanitized

## Screenshots/Screen recording (If applicable)

image
image

## Summary by CodeRabbit

* **New Features**
  * Form options can now be locked to prevent editing predefined multiple-choice answers.
  * Locked questions now show a clearer message when option editing is restricted.

* **Bug Fixes**
  * Registration now correctly treats “senior” as “adult” and rejects unsupported user type values instead of guessing.
  * Question controls are now disabled more consistently when a question is locked.

### CODE REVIEW

File: src/pages/FormBuilder/FormBuilder.tsx [hunk 1]
```diff
@@ -329,6 +332,7 @@ function InlineEditor({
           <label className={styles.inlineToggle}>
             <input
               type="checkbox"
+              disabled={question.locked}
               checked={question.required}
               onChange={(e) => onUpdate({ required: e.target.checked })}
```
[the remark this answers is not in the document]
Comment (evtran4) [reply src/pages/FormBuilder/FormBuilder.tsx:337]: Seems like a relevant issue. Please fix
